docusaurus/docs/dev-docs/configurations/typescript.md

```diff
@@ -1,11 +1,17 @@
 ---
-title: TypeScript configuration
+title: TypeScript and autocomplete
 displayed_sidebar: devDocsSidebar
 description: How to configure Strapi for TypeScript development. 
 
 ---
 
-# TypeScript project configuration
+# TypeScript and autocomplete configuration
+
+[TypeScript](/dev-docs/typescript)-enabled Strapi projects have a specific project structure and handle TypeScript project configuration through [`tsconfig.json` files](#project-structure-and-typescript-specific-configuration-files).
+
+Strapi also supports types autocompletion in your code editor while you are developing. This feature is related to TypeScript but works for both JavaScript and TypeScript projects. An additional [configuration parameter in the `config/typescript.js|ts` file](/dev-docs/configurations/typescript#strapi-specific-configuration-for-typescript-) allows Strapi to automatically generate types when content-types are updated.
```

Comment:
I like the fact that we mention the autogeneration here, but I believe types being exposed from Strapi packages is something that people will be expecting. Even if we're providing way more advanced types (including thanks to dynamic type registries & autogeneration), in the end, TypeScripts libraries are expected to provide auto-completion (including for JS apps). Thus I find it a bit weird that we advertise that as a feature.